src/azmi-main/blob/GetBlobs.cs

```diff
@@ -83,9 +100,9 @@ public List<string> Execute(Uri container, string directory, string identity = n
             logger.Debug($"Entering {className}::{MethodBase.GetCurrentMethod().Name}()");
 
             // authentication
-            var cred = new ManagedIdentityCredential(identity);
             Uri containerTrimmed = new Uri(container.ToString().TrimEnd(blobPathDelimiter));
-            var containerClient = new BlobContainerClient(containerTrimmed, cred);
+            var cred  = new ManagedIdentityCredential(identity);
+            containerClient ??= new ContainerClientImpl(containerTrimmed, cred);
```

Comment:
Null-coalescing assignment operator  is useless here and executes `if null` conditional. containerClient is always null here.
Well, maybe compiler does optimization under a hood, but I would rather get rid of this here.

Reply:
Nope, containerClient is **not** always null here. In unit testing we use constructor that prepopulates that property with mocked containerClient.